### PR TITLE
Add `TournamentListView` class for Enhanced Tournament Display

### LINKED ISSUE #137: Edit `views.py` for tournament
**Description:**

To better support our tournament features in the `src/chigame/games/views.py`:

- Introduce a new class, `TournamentListView` (or something similar) to handle tournament-specific listings.
- Implement queries within this class to retrieve relevant tournament matchups.
- Ensure the template context reflects the new structure for tournament matchups.

### PR DESCRIPTION
**Pull request by jb-01**
**Pull Request Description for Issue #137**

**Overview:**

This PR introduces enhancements to the `games/views.py` module in the `src/chigame` application, providing support for tournament features with a focus on listing and managing tournament-specific information.

**Key Changes:**

1. **`TournamentListView` Implementation:**
   - A new view class named `TournamentListView` has been added to handle the listing of tournaments.

2. **Tournament Matchup Queries:**
   - `TournamentListView` contains queries that fetch relevant tournament matchups. This allows for real-time display of active and forthcoming tournament games.

### CODE REVIEW

**Review by rqueiroz20 — CHANGES_REQUESTED** [review]
Great work with the tournament list view! The tournament page opens on my end and the migration went through as well. The one comment I will make is that I attempted to create a tournament object for its details to be displayed on the tournament page, but the tournament was not displayed. Perhaps I made a mistake when creating the object, but we should probably verify that tournament creation works before approving this PR.

**Review by elizabethli31 — CHANGES_REQUESTED** [review]
I'm also unable to view any tournaments. I think it could be because your context name is tournaments_detail, but you are feeding in tournaments? Also, please delete all the migrations from this PR along with the changes to managers.py. That's going to affect the migration merges for other teams once in dev.

**Comment by jb-01** [comment]
Thanks for the feedback. I have updated `views.py` to make the tournaments observable at http://127.0.0.1:8000/games/tournaments/. Please sign in as an admin to view the list.

File: src/chigame/games/views.py [hunk 1]
```diff
@@ -53,6 +53,34 @@ class GameEditView(UserPassesTestMixin, UpdateView):
...
+        context = super().get_context_data(**kwargs)
+        # Additional context can be added if needed
+        return context
+
     # check if user is staff member
     def test_func(self):
         return self.request.user.is_staff
+
+
+class TournamentCreateView(UserPassesTestMixin, CreateView):
```
Comment (elizabethli31) [inline src/chigame/games/views.py]: Is this supposed to be here? I thought Create was covered by PR #187

**Review by elizabethli31 — CHANGES_REQUESTED** [review]
List view is working now! I think you have the detail and create views on here which I believe is covered in another PR. Once you take those out of this PR, I can approve and merge.

**Comment by jb-01** [comment]
@elizabethli31 My latest commit resolves the additional views, and keeps this PR strictly related to Tournament List. The views, urls, and templates for `Create` and `Detail` were removed.

**Review by elizabethli31 — APPROVED** [review]
LGTM! Update with dev and I'll merge

**Comment by jb-01** [comment]
Thanks! I just updated with the Dev branch.

**Review by rqueiroz20 — APPROVED** [review]
The tournament list view is working properly now on my end! Thank you for addressing the comments made between myself and Elizabeth. Good job!

**Comment by jb-01** [comment]
Closing comment: I (jb-01) contributed to the TournamentListView in `views.py` which was heavily informed and guided by @MikeChen012345 and @MuyanXie contributions to `views.py` for Detail, Update, Delete, etc.

**Comment by elizabethli31** [comment]
Issue Score: Excellent

Comments:
Very nice work on creating a list view for tournaments. Also, great description, as it helped me identify exactly what I needed to test for the PR review.

**Pull request merged**